pyartcd/pyartcd/pipelines/quay_doomsday_backup.py

```diff
@@ -44,35 +48,66 @@ def sync_arch(self, arch: str):
             f"s3://ocp-doomsday-registry/release-image/{path}"
         ]
 
-        self.runtime.logger.info("Running mirror command: %s", mirror_cmd)
-        cmd_assert(mirror_cmd, retries=N_RETRIES)
-        self.runtime.logger.info("Mirror command ran successfully")
-        if self.runtime.dry_run:
-            self.runtime.logger.info("[DRY RUN] Would have run %s", " ".join(aws_cmd))
+        # Setup tenacity retry behavior for calling mirror_cmd and aws_cmd
+        # because cmd_assert_async does not have retry logic
+        retry = AsyncRetrying(reraise=True, stop=stop_after_attempt(N_RETRIES))
+        try:
+            self.runtime.logger.info("[%s] Running mirror command: %s", arch, mirror_cmd)
+            await retry(cmd_assert_async, mirror_cmd)
+            self.runtime.logger.info("[%s] Mirror command ran successfully", arch)
+            if self.runtime.dry_run:
+                self.runtime.logger.info("[DRY RUN] [%s] Would have run %s", arch, " ".join(aws_cmd))
+                self.runtime.logger.info("[DRY RUN] [%s] Would have messaged slack", arch)
+            else:
+                sleep(5)
```

Comment:
`time.sleep` will block the current coroutine. `await asyncio.sleep` should be used instead.

Reply:
Thank you! Fixed